patch/kernel/archive/rockchip64-6.12/board-rockpis-dts-fixes.patch

```diff
@@ -0,0 +1,130 @@
+From 3bdc6856ef82b6619c527034206b49a69f3d1b91 Mon Sep 17 00:00:00 2001
+From: Paolo Sabatino <paolo.sabatino@gmail.com>
+Date: Tue, 26 Nov 2024 12:13:24 +0100
+Subject: [PATCH] rk3308: fixes for rock pi s dts
+
+---
+ .../boot/dts/rockchip/rk3308-rock-pi-s.dts    | 75 +++++++++++++++++++
+ 1 file changed, 75 insertions(+)
+
+diff --git a/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts b/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts
+index 62d18ca769a1..8bcc57e9440c 100644
+--- a/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts
++++ b/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts
+@@ -48,6 +48,54 @@ blue-led {
+ 		};
+ 	};
+ 
++	analog_sound: analog-sound {
++		compatible = "simple-audio-card";
++		simple-audio-card,format = "i2s";
++		simple-audio-card,name = "analog";
++		simple-audio-card,mclk-fs = <256>;
++		simple-audio-card,codec-hp-det;
++		simple-audio-card,widgets =
++			"Headphone", "Headphones";
++
++		simple-audio-card,dai-link@1 {
++			format = "i2s";
++
++			cpu {
++				sound-dai = <&i2s_8ch_2>;
++			};
++
++			codec {
++				sound-dai = <&codec>;
++			};
++
++		};
++
++	};
++
++	pcm5102_sound: pcm5102-sound {
++		compatible = "simple-audio-card";
++		simple-audio-card,format = "i2s";
++		simple-audio-card,mclk-fs = <256>;
++		simple-audio-card,name = "pcm5102a";
++
++		simple-audio-card,dai-link@1 {
++			format = "i2s";
++			cpu {
++				sound-dai = <&i2s_8ch_0>;
++			};
++
++			codec {
++				sound-dai = <&pcm5102a>;
++			};
++		};
++	};
++
++	pcm5102a: pcm5102a {
++			#sound-dai-cells = <0>;
++			compatible = "ti,pcm5102a";
++			pcm510x,format = "i2s";
++	};
++
+ 	sdio_pwrseq: sdio-pwrseq {
+ 		compatible = "mmc-pwrseq-simple";
+ 		pinctrl-0 = <&wifi_enable_h>;
+@@ -128,6 +176,11 @@ vdd_log: vdd-log {
+ 	};
+ };
+ 
++&codec {
++	status = "okay";
++	#sound-dai-cells = <0>;
++};
++
+ &cpu0 {
+ 	cpu-supply = <&vdd_core>;
+ };
+@@ -139,6 +192,7 @@ &emmc {
+ 	non-removable;
+ 	pinctrl-names = "default";
+ 	pinctrl-0 = <&emmc_bus8 &emmc_clk &emmc_cmd>;
++	mmc-hs200-1_8v;
```

Comment:
The rock-pi-s and rock-s0 use a 3.3v io signal for emmc, adding this is possible incorrect hw description. According to emmc spec hs200 mode should use 1.8v io signal.

Does the SoM variant use 1.8v io signal for emmc?

Reply:
That is correct.  This is one of the reasons why I was concerned when you proposed enabling it.
But, the driver seems to ignore it on hardware that will not support the higher rates.
Do you have 1.8v I/O to your EMMC on the core board?
If so, have you actually observed it running faster with the HS200 enabled?

Reply:
If none of these little boards really supports HS200, maybe we should remove it so as not to give folks false hope :-)

Reply:
@brentr that's me that I introduced the hs200 property in a past pull request: https://github.com/armbian/build/pull/7463

The SoM module and its emmc supports hs200, also according to the schematic on https://dl.radxa.com/rockpis/docs/hw/core/ROCKPI_S_CORE_sch_v1.1_20200320.pdf on RK3308 Part-B section, there is `FLASH_SEL` which select 1.8v when high, and the pin should be already set as pull-up by default.

I guess this is a good argument and one step forward to have a separate Rock Pi S Core SoM board config.

Reply:
According to the schematic the pin is connected to ground, i.e. low and that signal 3.3v usage.

The vccio3 io-domain is also using vcc_io regulator that is 3.3v, and that is signaled and configured in io-domain node.

I would be careful trying to enable hs200 on all rock-pi-s boards.

Reply:
You're right, I didn't notice the pin goes to ground on the schematic :sleeping: 

Anyway, all the rock pi s revisions seems to have GPIO0_A4 bound to `vccio_flash` from the schematics, so I guess in theory a good idea is to add the gpio regulator and wire it to mmc controller, but since the older rock pi s have the sdnand thingy probably is a good idea to split and derive a Core SoM specific DTS from the main rock pi s boards to avoid any nuissances (there's the 4 bit bus too...)

anyway @brentr checked on his board and the kernel was able to autodetect the bus size and the missing hs200 modes, but to be on the safe side probably it is better to move away.

Reply:
I guess I wrong also wrong about the regulator (sorry, no electronic engineer here): the resistor is low enough (2.2kohm) so I guess it is a pull-down resistor made to keep the pin down. The GPIO0_A4 line at this point is not supposed to be turned into an output GPIO, but rather to an input GPIO to signal the kernel/driver at which voltage the vccio-flash is hardwired.

Reading the GPIO0_A4 tells the line is low according to the schematic:
```
root@rockpi-s:/sys/class/gpio# echo 4 > export 
root@rockpi-s:/sys/class/gpio# cat gpio4/direction 
in
root@rockpi-s:/sys/class/gpio# cat gpio4/value 
0
```

so I removed the `mmc-hs200-1_8v` property from the dtb.

note: I added, for couriousity, the `mmc-ddr-3_3v` property, and the eMMC goes in DDR mode correctly. It is way slower than hs200 (83MB/s vs. 130MB/s), but the DDR mode should be within the specs.